pkg/controller/v1beta1/inferenceservice/components/component.go

```diff
@@ -13,9 +13,90 @@ limitations under the License.
 
 package components
 
-import "github.com/kubeflow/kfserving/pkg/apis/serving/v1beta1"
+import (
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1beta1"
+	"github.com/kubeflow/kfserving/pkg/constants"
+	"github.com/kubeflow/kfserving/pkg/controller/v1alpha1/trainedmodel/sharding/memory"
+	v1beta1utils "github.com/kubeflow/kfserving/pkg/controller/v1beta1/inferenceservice/utils"
+	v1 "k8s.io/api/core/v1"
+	"strconv"
+)
 
 // Component can be reconciled to create underlying resources for an InferenceService
 type Component interface {
 	Reconcile(isvc *v1beta1.InferenceService) error
 }
+
+func addLoggerAnnotations(logger *v1beta1.LoggerSpec, annotations map[string]string) bool {
+	if logger != nil {
+		annotations[constants.LoggerInternalAnnotationKey] = "true"
+		if logger.URL != nil {
+			annotations[constants.LoggerSinkUrlInternalAnnotationKey] = *logger.URL
+		}
+		annotations[constants.LoggerModeInternalAnnotationKey] = string(logger.Mode)
+		return true
+	}
+	return false
+}
+
+func addLoggerContainerPort(container *v1.Container) {
```

Comment:
`addLoggerContainerPort` and `addBatcherContainerPort` is the same function now as they are all merged to agent container, can you merge these to `addAgentContainerPort`?

Reply:
0cfefe9